src/api/spec/validators/workflow_event_filter_validator_spec.rb

```diff
@@ -0,0 +1,60 @@
+require 'rails_helper'
+
+RSpec.describe WorkflowEventFilterValidator do
+  describe '#validate' do
+    let(:yaml) do
+      {
+        'steps' => [{ 'branch_package' => {
+          'source_project' => 'test-project',
+          'source_package' => 'test-package',
+          'target_project' => 'test-target-project'
+        } }],
+        'filters' => filters
+      }
+    end
+
+    context 'there are no supported filters' do
+      let(:filters) { { 'event' => 'foo' } }
+
+      subject { Workflow.new(workflow_instructions: yaml) }
+
+      it 'is not valid and has an error message' do
+        expect(subject).not_to(be_valid)
+        expect(subject.errors.full_messages.to_sentence).to eql('Workflow filter not supported: foo')
+      end
+    end
+
+    context 'there are no event filters' do
+      let(:filters) { {} }
+
+      subject { Workflow.new(workflow_instructions: yaml) }
+
+      it 'is not valid and has an error message' do
+        expect(subject).not_to(be_valid)
```

Comment:
This will also call all other validation code of the `Workflow` model. So even if the custom validator doesn't change, its specs could break due to changes in other validation code. I suggest using a fake model like we did in [this spec](https://github.com/openSUSE/open-build-service/blob/4ce0dfe110d023e90c3673fa9d673391490425e8/src/api/spec/validators/workflow_filters_validator_spec.rb#L4-L10) for another custom validator.